coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/CoralToTrinoSqlCallConverter.java

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright 2017-2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.trino.rel2trino;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableList;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.sql.util.SqlShuttle;
+
+import com.linkedin.coral.com.google.common.base.CaseFormat;
+import com.linkedin.coral.com.google.common.base.Converter;
+import com.linkedin.coral.com.google.common.collect.ImmutableMultimap;
+import com.linkedin.coral.common.functions.Function;
+import com.linkedin.coral.common.transformers.OperatorBasedSqlCallTransformer;
+import com.linkedin.coral.common.transformers.SqlCallTransformer;
+import com.linkedin.coral.common.transformers.SqlCallTransformers;
+import com.linkedin.coral.hive.hive2rel.functions.HiveRLikeOperator;
+import com.linkedin.coral.hive.hive2rel.functions.StaticHiveFunctionRegistry;
+import com.linkedin.coral.trino.rel2trino.functions.TrinoElementAtFunction;
+import com.linkedin.coral.trino.rel2trino.transformers.LinkedInOperatorBasedSqlCallTransformer;
+import com.linkedin.coral.trino.rel2trino.transformers.MapStructAccessOperatorTransformer;
+import com.linkedin.coral.trino.rel2trino.transformers.ToDateOperatorTransformer;
+
+import static com.linkedin.coral.trino.rel2trino.CoralTrinoConfigKeys.*;
+
+
+/**
+ * This class extends the class of SqlShuttle and initialize a SqlCallTransformers which containing a list of SqlCallTransformers
+ * to traverse the hierarchy of a SqlCall and converts the functions from Coral operator to Trino operator if it is required
+ */
+public class CoralToTrinoSqlCallConverter extends SqlShuttle {
+  public static final StaticHiveFunctionRegistry HIVE_FUNCTION_REGISTRY = new StaticHiveFunctionRegistry();
+  private final Map<String, Boolean> configs;
+  private final SqlCallTransformers sqlCallTransformers;
+  private final Set<String> linkedInFunctionSignatureSet;
+
+  private final List<SqlCallTransformer> sqlCallTransformerList;
+  public CoralToTrinoSqlCallConverter(Map<String, Boolean> configs) {
+    this.configs = configs;
+    this.linkedInFunctionSignatureSet = new HashSet<>();
+    this.sqlCallTransformerList = new ArrayList<>();
+
+    // conditional functions
+    sqlCallTransformerList.add(new OperatorBasedSqlCallTransformer(hiveToCoralSqlOperator("nvl"), 2, "coalesce"));
```

Comment:
Here you are converting from name to operator then calling a constructor that converts operator to name.